app/assets/javascripts/discourse-markdown-it/src/features/text-post-process.js

```diff
@@ -59,7 +61,8 @@ export class TextPostProcessRuler {
         this.rules[i].rule.onMatch(
           buffer,
           match.slice(index, this.matcherIndex[i + 1]),
-          state
+          state,
+          { parseBBCodeTag, applyDataAttributes }
```

Comment:
This is what I did to remove the `shims`. We are now passing both `applyDataAttributes` and `parseBBCode` function to the recipient of the `onMatch` function so they can use it to do their parsing.